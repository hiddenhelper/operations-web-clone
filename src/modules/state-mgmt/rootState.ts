import { History } from 'history';

import { IState as IAuthState } from './auth/state';
import { IState as IGeneralState } from './general/state';
import { IState as IClientState } from './client/state';
import { IState as IProjectState } from './project/state';
import { IState as IUserState } from './user/state';
import { IState as IAccessControlSystemState } from './access-control-system/state';
import { IState as IBadgePrintingSystemState } from './badge-printer-system/state';
import { IState as IWorkerState } from './worker/state';
import { IState as ICertificationState } from './certification/state';
import { IState as IFileState } from './file/state';
import { IState as IBadgeState } from './badge/state';
import { IState as IStatisticsState } from './statistics/state';
import { IState as ITrainingState } from './training/state';
import { IState as IPaymentState } from './payment/state';
import { IState as IInvoiceState } from './invoice/state';
import { ApiService } from '../services/ApiService';
import { AuthService } from '../services/AuthService';
import { PrinterService } from '../services/PrinterService';
import { FileService } from '../services/FileService';
import { Logger } from '../services/Logger';
import { IProcoreState } from './procore/state';
import { IState as IProjectNewState } from './project-new/state';

export interface IAction {
  type: string;
  payload: any;
}

export interface IRootState {
  auth: IAuthState;
  user: IUserState;
  client: IClientState;
  project: IProjectState;
  accessControlSystem: IAccessControlSystemState;
  badgePrinterSystem: IBadgePrintingSystemState;
  worker: IWorkerState;
  certification: ICertificationState;
  training: ITrainingState;
  invoice: IInvoiceState;
  file: IFileState;
  badge: IBadgeState;
  statistics: IStatisticsState;
  payment: IPaymentState;
  general: IGeneralState;
  router?: { location: { pathname: string; search: string; hard: string; key: string }; action: string };
  procore: IProcoreState;
  projectNew: IProjectNewState;
}

export interface IEpicDependencies {
  history: History;
  apiService: ApiService;
  authService: AuthService;
  printerService: PrinterService;
  fileService: FileService;
  logger: typeof Logger;
}
