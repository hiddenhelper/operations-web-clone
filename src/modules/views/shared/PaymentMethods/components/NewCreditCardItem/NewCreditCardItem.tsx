import React, { memo, useCallback, useState } from 'react';
import { useStyles } from '../../../PaymentForm/styles';
import Button from '@material-ui/core/Button';
import PaymentForm from '../../../PaymentForm/PaymentForm';
import { CloseIcon, CreditCardIcon, LANG } from '../../../../../../constants';
import { useStyles as modalStyles } from '../../../Modal/style';
import Typography from '@material-ui/core/Typography';

const NewCreditCardItem = () => {
  const classes = useStyles();
  const modalClasses = modalStyles();
  const [isNew, setIsNew] = useState(true);

  const addNewCreditCard = useCallback(() => {
    setIsNew(false);
  }, []);

  const handleNewCardAdded = useCallback(() => {
    setIsNew(true);
  }, []);

  return (
    <div className={`${isNew && classes.emptyCardItem} ${!isNew && classes.formCardItem}`}>
      {isNew && (
        <Button className={`${classes.addNewCardButton}`} onClick={addNewCreditCard} color="primary" data-testid="add-card-btn">
          <CreditCardIcon />
          ADD NEW CREDIT CARD
        </Button>
      )}
      {!isNew && (
        <>
          <div className={classes.newCardContainer}>
            <Button
              disableRipple={true}
              className={`${modalClasses.closeButton} ${classes.closeButton}`}
              data-testid="close-credit-card-form"
              onClick={handleNewCardAdded}
            >
              <CloseIcon />
            </Button>
            <Typography className={classes.cardFormTitle} color="primary" align="left" component="h1" variant="h5">
              Enter Credit Card Information
            </Typography>
            <Typography className={classes.cardFormSubtitle}>{LANG.EN.PAYMENT.CREDIT_CARD_FORM_DESCRIPTION}</Typography>
            <div className={classes.newCardFormContainer}>
              <div>
                <PaymentForm onSuccessCallback={handleNewCardAdded} />
              </div>
            </div>
          </div>
        </>
      )}
    </div>
  );
};

export default memo(NewCreditCardItem);
