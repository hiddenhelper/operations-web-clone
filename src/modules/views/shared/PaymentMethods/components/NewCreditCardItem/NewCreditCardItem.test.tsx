import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';
import { getClientAdminInitialState } from '../../../../../../test/rootState';
import { MemoryRouter } from 'react-router';
import NewCreditCardItem from './NewCreditCardItem';

describe('New Credit Card Item', () => {
  it('should render', () => {
    const wrapper = render(
      <Provider store={createMockStore(getClientAdminInitialState())}>
        <MemoryRouter>
          <NewCreditCardItem />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render credit card form', () => {
    const wrapper = render(
      <Provider store={createMockStore(getClientAdminInitialState())}>
        <MemoryRouter>
          <NewCreditCardItem />
        </MemoryRouter>
      </Provider>
    );

    act(() => {
      fireEvent.click(wrapper.getByText('ADD NEW CREDIT CARD'));
    });

    expect(wrapper.container).toMatchSnapshot();

    act(() => {
      fireEvent.click(wrapper.getByTestId('close-credit-card-form'));
    });

    expect(wrapper.container).toMatchSnapshot();
  });
});
