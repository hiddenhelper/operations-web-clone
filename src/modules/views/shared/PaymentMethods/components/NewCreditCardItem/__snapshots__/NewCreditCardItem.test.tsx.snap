// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Credit Card Item should render 1`] = `
<div>
  <div
    class="makeStyles-emptyCardItem-1 false"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addNewCardButton-23 MuiButton-textPrimary"
      data-testid="add-card-btn"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <svg>
          credit-card.svg
        </svg>
        ADD NEW CREDIT CARD
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`New Credit Card Item should render credit card form 1`] = `
<div>
  <div
    class="false makeStyles-formCardItem-71"
  >
    <div
      class="makeStyles-newCardContainer-54"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-closeButton-80 makeStyles-closeButton-73"
        data-testid="close-credit-card-form"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <svg>
            close.svg
          </svg>
        </span>
      </button>
      <h1
        class="MuiTypography-root makeStyles-cardFormTitle-56 MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-alignLeft"
      >
        Enter Credit Card Information
      </h1>
      <p
        class="MuiTypography-root makeStyles-cardFormSubtitle-57 MuiTypography-body1"
      >
        Please enter your Credit Card details. This card will be charged every payment cycle (weekly or monthly) as long as it is associated with a Project.
      </p>
      <div
        class="makeStyles-newCardFormContainer-55"
      >
        <div>
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`New Credit Card Item should render credit card form 2`] = `
<div>
  <div
    class="makeStyles-emptyCardItem-50 false"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addNewCardButton-72 MuiButton-textPrimary"
      data-testid="add-card-btn"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <svg>
          credit-card.svg
        </svg>
        ADD NEW CREDIT CARD
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
